forecasting/setup.py

```diff
@@ -26,10 +28,17 @@
     "pandas",
     "psutil>=5.7.3,<5.9",
     "gluonts>=0.8.0",
-    "mxnet<2.0",
```

Comment:
What would happen if we released forecasting today and had it install when `pip install autogluon` was called? Would the user get an error during install if they didn't have MXNet installed prior? If so, why error instead of installing MXNet automatically? Do any of forecasting's dependencies forcibly install MXNet?

Reply:
For v0.5 release, are we planning to continue having a hard dependency on MXNet? If so we must choose one of the following options:

1. Consider not having forecasting install with `pip install autogluon` and instead request the user to specifically install `pip install autogluon.forecasting`.
2. Install forecasting with autogluon and don't auto-install mxnet, then if the user tries to import forecasting they get an error if they didn't have mxnet installed which tells them to install mxnet. This is what currently happens with autogluon's object detector which only works with MXNet.
3. (Not Recommended) Forecasting installs with autogluon, forcibly installs mxnet when calling `pip install autogluon`.

Open to hearing your thoughts.

Reply:
also tagging @gradientsky as this PR grew from a discussion with him.

Yes, if we released like this, installation would fail if mxnet is not installed. Also, I don't think any of the dependencies forcibly install it. The reason is as follows.

`mxnet`, in PyPI, installs the library without CUDA bindings, defaulting to no GPU usage. If we include `mxnet` in dependencies, it reinstalls the library in a CPU-only mode, which was actually the issue uncovered by CI in this PR. CUDA-ready versions go by `mxnet-cu***`.

Configured like this, users are expected to bring their own mxnet, often via a container configuration. This is also the path followed by GluonTS. See their [README](https://github.com/awslabs/gluon-ts).

For the v0.5 we will definitely need mxnet for healthy operation. I'd prefer option 2. I am good with option 1, where we can ask the user to install with 

```pip install autogluon.forecasting mxnet~=1.7```

Another path we might take is to put more time into actually averting mxnet altogether, and depending on only pytorch models in GluonTS. This would shrink the model zoo in the interim.

Reply:
+1 on option 2. These binaries are also pre-installed in DL AMI (which one depends on instance type and if supports GPU).